strings/boyer_moore.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file
+ * @brief Boyer-moore's algorithm finding all occurrences of pattern in given
+ * text.
+ * @author [Stoycho Kyosev](https://github.com/stoychoX)
+ */
+#include <cassert>
+#include <cstring>
+#include <iostream>
+#include <limits>
+#include <vector>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```